### PR TITLE
Follow thenable state not own state

### LINKED ISSUE #101: Follow thenable state, not own state
This code looks suspicious:

```
function handleOwnThenable(promise, thenable) {
  if (thenable._state === FULFILLED) {
    fulfill(promise, thenable._result);
  } else if (promise._state === REJECTED) {
    reject(promise, thenable._result);
  } else {
    subscribe(thenable, undefined, function(value) {
      resolve(promise, value);
    }, function(reason) {
      reject(promise, reason);
    });
  }
}
```

Seems like it [has been changed in RSVP](https://github.com/tildeio/rsvp.js/commit/409a3557012535c949a7fb6f488c760c59656d43). Should we sync it ? 

May be there should be some unit test for that.

Edit: see the initial PR for [comments on tests](https://github.com/tildeio/rsvp.js/pull/333)

### PR DESCRIPTION
**Pull request by vicb**
fix #101 

This is only syncing a commit from [rsvp](https://github.com/tildeio/rsvp.js/pull/333), credits to @teddyzeenny for the original changes

### CODE REVIEW

File: lib/es6-promise/-internal.js [hunk 1]
```diff
@@ -66,7 +66,8 @@ function handleForeignThenable(promise, thenable, then) {
 function handleOwnThenable(promise, thenable) {
   if (thenable._state === FULFILLED) {
     fulfill(promise, thenable._result);
-  } else if (promise._state === REJECTED) {
+  } else if (thenable._state === REJECTED) {
+    thenable._onError = null;
```
Comment (stefanpenner) [inline lib/es6-promise/-internal.js]: This little line isn't needed

Comment (vicb) [inline lib/es6-promise/-internal.js]: I had no idea why it was there but it is in the rsvp commit.

Should we add a comment on why the impl is different from rsvp ?

**Comment by stefanpenner** [comment]
This library doesn't have the same instrumentation hooks. And that is one of the hooks

**Comment by vicb** [comment]
@stefanpenner thanks for the explanation, makes perfect sense now. PR updated.

**Comment by stefanpenner** [comment]
:+1:

**Pull request merged**